cparser/Cutil.ml

```diff
@@ -922,7 +925,9 @@ let binary_conversion env t1 t2 =
   | TFloat(FDouble, _), (TInt _ | TFloat _)     -> t1
   | (TInt _ | TFloat _), TFloat(FDouble, _)     -> t2
   | TFloat(FFloat, _), (TInt _ | TFloat _)      -> t1
-  | (TInt _), TFloat(FFloat, _)      -> t2
+  | (TInt _ | TFloat _), TFloat(FFloat, _)      -> t2
+  | TFloat(FFloat16, _), (TInt _ | TFloat _)    -> t1
```

Comment:
This line confused me.

    | TFloat(FFloat16, _), TInt _ -> t1

This is easier to understand.

Reply:
But then `TFloat(FFloat16, _), TFloat(FFloat16, _)` is not matched by any pattern, I'm afraid...

I agree this code is confusing, and I have some ideas on how to clean it up, but the addition of Float16 in this PR follows the same pattern as the current code, just with more cases.

Reply:
Yes, OK.